code/controlconfig/controlsconfigcommon.cpp

```diff
@@ -1156,6 +1168,108 @@ size_t find_control_by_text(SCP_string &text) {
 	return item_id;
 }
 
+/*
+* Button hooks can happen more than once a frame, and that is intended.
+* Axis hooks happen exactly once a frame.
+* But due to implementation, continuous hooks could happen an unspecified number of times. So we need to cache if a hook occurred, and what it's override setting was
+* Bit at IoActionId * 2 sets if this action has been run before, Bit at IoActionId * 2 + 1 sets the cached value
+*/
+std::bitset<IoActionId::CCFG_MAX * 2> Controls_lua_override_cache;
```

Comment:
This looks good, but it's probably a good idea to wrap the bitset with a class so its users won't need to be careful about which bit they're accessing.

Reply:
What that being the case... it may be advtantageous to actually add these bools to the CCI's themselves.

Reply:
This cache is only, and should only ever be used for continuous buttons, which can get queried more than once per evaluation period.
Axes will get queried once a frame and don't need this cache and normal buttons must be able to be called exactly when it's underlying engine code would run (which it currently does).
Hence I'd like to keep this one separate as well, and, if it really needs to be a class, would like to just create an anonymous singleton class here (no other file should also ever really access this)

Reply:
As discussed on it was decided to use a bitfield (now wrapped in a singleton class for usage security) to hold lua-specific states of the controls, in contrast to lumping them together with the other bool in the CCI.  This was done on the basis that bool packing within classes is not a C++ standard, and is OS specific, and since we're doing controls routine that's basically firmware level, we need to consider performance and size constraints.